fgpyo/sam/__init__.py

```diff
@@ -547,6 +547,40 @@ def length_on_target(self) -> int:
         """Returns the length of the alignment on the target sequence."""
         return sum([elem.length_on_target for elem in self.elements])
 
+    def get_alignment_offsets(self, reverse: bool = False) -> Tuple[int, int]:
+        """
+                Get the starting and ending offset for the alignment based on the CIGAR string.
+
+                Args:
+                    reverse: Whether to count from the end of the read sequence.
+                        Default is False, which counts from the beginning of the read sequence.
+
+                Returns: Tuple[int, int] The start and end of the _aligned part_ of the read.
+                    0-based, open-ended offsets (to the read sequence.)
+                    If 'reverse' is True, the offsets count from the end of the read sequence.
```

Comment:
```suggestion
        Get the starting and ending offsets for the alignment based on the CIGAR string.

        Args:
            reverse: If True, count from the end of the read sequence.
                Otherwise (default behavior), count from the beginning of the read sequence.

        Returns: 
            A tuple (start, end), defining the start and end offsets of the _aligned part_ of the read.
            These offsets are 0-based and open-ended, with respect to the beginning of the read sequence.
            (If 'reverse' is True, the offsets are with respect to the end of the read sequence.)
```

**nit** please adhere to google style when documenting a tuple return value

> Do not imitate older ‘NumPy style’ ([example](https://numpy.org/doc/1.24/reference/generated/numpy.linalg.qr.html)), which frequently documented a tuple return value as if it were multiple return values with individual names (never mentioning the tuple). Instead, describe such a return value as: “Returns: A tuple (mat_a, mat_b), where mat_a is …, and …”. The auxiliary names in the docstring need not necessarily correspond to any internal names used in the function body (as those are not part of the API)

https://google.github.io/styleguide/pyguide.html

**nit** when documenting a boolean parameter, I always try to be explicit about the behavior when the bool is set to True (or False), rather than a phrase beginning with "Whether", which can be ambiguous

**nit** also, please dedent so the text is inline with the triple-quote